src/AgGrid.elm

```diff
@@ -833,21 +1026,22 @@ generateGridConfigAttributes gridConfig =
                         Json.Encode.null
               )
             , ( "sideBar"
-              , case gridConfig.sideBar of
-                    BothSidebars ->
-                        Json.Encode.bool True
-
-                    ColumnSidebar ->
-                        Json.Encode.string "columns"
-
-                    FilterSidebar ->
-                        Json.Encode.string "filters"
-
-                    NoSidebar ->
-                        Json.Encode.bool False
+              , Json.Encode.object
+                    [ ( "toolPanels", Json.Encode.list encodeSidebarType gridConfig.sideBar.panels )
+                    , ( "hiddenByDefault", Json.Encode.bool gridConfig.sideBar.hiddenByDefault )
+                    , ( "position"
+                      , case gridConfig.sideBar.position of
+                            SidebarLeft ->
+                                Json.Encode.string "left"
+
+                            SidebarRight ->
+                                Json.Encode.string "right"
+                      )
+                    , ( "defaultToolPanel", encodeSidebarType gridConfig.sideBar.defaultToolPanel )
```

Comment:
So `hiddenByDefault` doesn't really do anything for us because the user can just set `panels = []` to hide the sidebar.

I will remove `hiddenByDefault` and change the `defaultToolPanel` to a `Maybe`.